src/daemon.rs

```diff
@@ -226,6 +239,17 @@ async fn peel_forward_loop(ctx: DaemonContext) -> anyhow::Result<()> {
     }
 }
 
+async fn dispatch_by_dock_loop(ctx: DaemonContext) -> anyhow::Result<()> {
+    loop {
+        if let Some((message, fingerprint)) = ctx.recv_message().await {
+            match ctx.socket_recv_queues.get(message.get_dest_dock()) {
+                Some(sender) => sender.send((message, fingerprint)).await?,
```

Comment:
What happens when the receive queue for this socket is full, and nobody is calling recv_from on the other end?

Using `try_send` can avoid blocking this entire loop. Dropping packets occasionally is fine.